[Liu Yujia] iP

LiuYJ2002 opened the pull request:
# **DukePro**
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,
-text-based
-easy to learn
-FAST SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LiuYJ2002 committed:
level 0

LiuYJ2002 committed:
Level 1

LiuYJ2002 committed:
level 2

LiuYJ2002 committed:
Level 3

LiuYJ2002 committed:
Level 4

LiuYJ2002 committed:
testing

LiuYJ2002 committed:
fixed todo bugs for level 4

LiuYJ2002 committed:
modified testing

LiuYJ2002 committed:
Level 5

LiuYJ2002 committed:
Level 6

LiuYJ2002 committed:
Update Daniel to save task and load task from hard disk

LiuYJ2002 committed:
Merge branch 'branch-level-7'

LiuYJ2002 committed:
Add date time recognition functionality to chatbot

LiuYJ2002 committed:
Merge branch 'branch-level-8'

LiuYJ2002 committed:
Add more OOP into program

LiuYJ2002 committed:
Add packaging to all files

LiuYJ2002 committed:
Merge branch 'branch-A-packages'

LiuYJ2002 committed:
Merge branch 'add-gradle-support'

LiuYJ2002 committed:
Add JUnit tests

LiuYJ2002 committed:
Merge branch 'A-JUnit'

LiuYJ2002 committed:
Create jar file

LiuYJ2002 committed:
Add javadoc

LiuYJ2002 committed:
Merge branch 'branch-A-JavaDoc'

LiuYJ2002 committed:
Tweak code to comple with a coding standard

LiuYJ2002 committed:
Add find keyword functionality

LiuYJ2002 committed:
Merge branch 'branch-A-CodingStandard'

LiuYJ2002 committed:
Resolve merge conflict

yiiilonggg commented on src/main/java/Ui/Ui.java:
```diff
@@ -0,0 +1,44 @@
+            i += 1;
+        }
+    }
+    public static void uiTodo(int size, Todo x) {
```
For the naming of the Ui methods, should they be of the verb form instead?

yiiilonggg commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+    }
+    public void search(String input) {
+        String matchingWord = Parse.parseFind(input);
+        List<Task> match = new ArrayList<>();
```
for the naming of this variable, should the plural form be used?

yiiilonggg reviewed (commented):
Overall, the code quality guidelines were mostly followed!

Sufficient documentation for non-trivial methods 👍 
Variable and method names are not misleading 👍 

Great job!

JovenSoh reviewed (commented):
Excellent adherence to coding standards throughout the project. Code is clean, well-structured, and easy to read. You've done a great job in maintaining consistent formatting, and your adherence to the project conventions is evident.

Well-documented non-trivial methods. Javadoc comments provide clear explanations, particularly for more complex methods and classes. Keep up the great documentation practices!

JovenSoh commented on src/main/java/Storage/Storage.java:
For the import statements, would it be better to group the Task imports together? Also, we should aim to explicitly import classes instead of using wildcards

LiuYJ2002 committed:
Add GUI to daniel

Elinengu commented on src/main/java/Ui/Ui.java:
```diff
@@ -3,42 +3,65 @@
         System.out.println("Bye hope to see you again soon");
+        return s;
     }
-    public static void uiList(List<Task> array) {
```
Perhaps it would be better to give a more meaningful name such as list instead of array?

Elinengu commented on src/main/java/Ui/Ui.java:
```diff
@@ -3,42 +3,65 @@
 import java.util.List;
 import Task.*;
```
perhaps it's better not to import entire package using *? as it violates the coding standard

Elinengu commented on src/main/java/Task/TaskList.java:
```diff
@@ -62,43 +62,43 @@ public void handleTask(String input) throws WrongKeyword, MissingArg {
-        Ui.uiDelete(t, array.size());
+        return Ui.uiDelete(t, array.size());
     }
     public List<Task> getArray() {
```
maybe it would be better to rename the method as getList()?

LiuYJ2002 committed:
Add assertions to Daniel

This commit adds Java assertions to certain parts of code
to document and test certain assumptions in the code during
production.

LiuYJ2002 committed:
Improve code code quality

Code quality is improved by removing magic literals and numbers.
Unused code is also remove to simplify it and loops are shortened.

LiuYJ2002 committed:
Merge pull request #1 from LiuYJ2002/branch-A-Assertions

Add assertions to Daniel

LiuYJ2002 committed:
Merge branch 'branch-A-CodeQuality' into master

LiuYJ2002 committed:
Merge pull request #2 from LiuYJ2002/master

A-CodeQuality

LiuYJ2002 committed:
Add CI using github actions

LiuYJ2002 committed:
Merge branch 'branch-A-CI'

LiuYJ2002 committed:
fixed bugs in github workflows

LiuYJ2002 committed:
Add sorting function to sort according to deadline

LiuYJ2002 committed:
Merge branch 'branch-BCD-extension'

LiuYJ2002 committed:
Merge pull request #4 from LiuYJ2002/master

Second time fixing difference from previous failed merge

LiuYJ2002 committed:
Merge pull request #5 from LiuYJ2002/branch-A-CodeQuality

Branch a code quality

LiuYJ2002 committed:
Add small UI changes

LiuYJ2002 committed:
Improve Gui to Daniel

LiuYJ2002 committed:
Merge branch 'branch-A-BetterGui'

LiuYJ2002 committed:
Improve error handling

LiuYJ2002 committed:
Add Ui png file to docs folder

LiuYJ2002 committed:
Update Readme file

LiuYJ2002 committed:
Merge branch 'branch-A-UserGuide'

LiuYJ2002 committed:
Fix code styles and javadoc

LiuYJ2002 committed:
Improve JUnit tests

LiuYJ2002 committed:
Resolve minor bugs

LiuYJ2002 committed:
changed readme file
